src/main/java/in/koreatech/koin/domain/owner/service/OwnerService.java

```diff
@@ -127,6 +128,30 @@ public void register(OwnerRegisterRequest request) {
         eventPublisher.publishEvent(new OwnerRegisterEvent(saved));
     }
 
+    @Transactional
+    public void registerByPhone(OwnerRegisterByPhoneRequest request) {
+        if (userRepository.findByPhoneNumberAndUserType(request.phoneNumber(), OWNER).isPresent()) {
+            throw DuplicationPhoneNumberException.withDetail("phoneNumber: " + request.phoneNumber());
+        }
+        if (ownerRepository.findByCompanyRegistrationNumber(request.companyNumber()).isPresent()) {
+            throw DuplicationCompanyNumberException.withDetail("companyNumber: " + request.companyNumber());
+        }
+        Owner owner = request.toOwner(passwordEncoder);
+        Owner saved = ownerRepository.save(owner);
+        if (request.shopId() != null) {
+            var shop = shopRepository.getById(request.shopId());
+            ownerShopRedisRepository.save(OwnerShop.builder()
+                .ownerId(owner.getId())
+                .shopId(shop.getId())
+                .build());
+        } else {
+            ownerShopRedisRepository.save(OwnerShop.builder()
+                .ownerId(owner.getId())
+                .build());
+        }
```

Comment:
```suggestion
 OwnerShop.OwnerShopBuilder ownerShopBuilder = OwnerShop.builder().ownerId(owner.getId());
    if (request.shopId() != null) {
        var shop = shopRepository.getById(request.shopId());
        ownerShopBuilder.shopId(shop.getId());
    }
    ownerShopRedisRepository.save(ownerShopBuilder.build());
```
ownerShopRedisRepository.save(OwnerShop.builder()가 중복으로 쓰이는 거 같아서 위와같은 코드는 어떨까요?

Reply:
좋습니다~